examples/quantization/brevitas/quantize_llm.py

```diff
@@ -95,40 +106,42 @@
     nsamples=128,
     seqlen=args.seqlen,
     split="validation",
-    device="cuda:0" if not args.cpu_offload else None,
+    device=args.device if args.device != "auto" else None,
 )
 
-perplexity = compute_perplexity(
-    quantizer.model, validation_dataset, context_length=args.seqlen // 2, tokenizer=tokenizer
-)
+model = quantizer.model
+if args.device != "auto":
+    offload_context = contextlib.nullcontext()
+else:
+    offload_context = accelerate_offload(model, qconfig.gpu_device_map, qconfig.cpu_device_map)
 
+# Evaluation of the non-quantized model.
+with offload_context:
+    perplexity = compute_perplexity(model, validation_dataset, context_length=args.seqlen // 2, tokenizer=tokenizer)
 print(f"Perplexity (original model): {perplexity}")
 
-quantized_model = quantizer.quantize(qconfig, calibration_dataset)
+model = quantizer.quantize(qconfig, calibration_dataset)
+if args.device == "auto":
+    offload_context = accelerate_offload(model, qconfig.gpu_device_map, qconfig.cpu_device_map)
 
 # Evaluation of the quantized model.
-perplexity = compute_perplexity(
-    quantized_model, validation_dataset, context_length=args.seqlen // 2, tokenizer=tokenizer
-)
+with offload_context:
+    perplexity = compute_perplexity(model, validation_dataset, context_length=args.seqlen // 2, tokenizer=tokenizer)
 print(f"Perplexity (quantized model): {perplexity}")
 
 print("Exporting the model to ONNX...")
-if args.cpu_offload:
-    # When exporting to ONNX, Accelerate's hooks need to be removed otherwise we have unwanted Cast nodes in the ONNX graph.
-    remove_hooks(quantized_model)
-
-quantized_model = quantized_model.to("cpu")
-for name, param in quantized_model.named_parameters():
+model = model.to("cpu")
+for name, param in model.named_parameters():
     if param.dtype in [torch.float16, torch.bfloat16]:
-        recurse_setattr(quantized_model, name, torch.nn.Parameter(param.to(torch.float32)))
-for name, param in quantized_model.named_buffers():
+        recurse_setattr(model, name, torch.nn.Parameter(param.to(torch.float32)))
+for name, param in model.named_buffers():
     if param.dtype in [torch.float16, torch.bfloat16]:
-        recurse_setattr(quantized_model, name, torch.nn.Parameter(param.to(torch.float32)))
+        recurse_setattr(model, name, torch.nn.Parameter(param.to(torch.float32)))
```

Comment:
Let's remove this bit (we'll handle `torch_dtype` later - I shouldn't have added it in the first place)